app/models/mixins/authentication_mixin.rb

```diff
@@ -70,6 +70,10 @@ def authentication_key_pairs
     authentications.select { |a| a.kind_of?(ManageIQ::Providers::Openstack::InfraManager::AuthKeyPair) }
   end
 
+  def authentication_private_keys
+    authentications.select { |a| a.kind_of?(AuthPrivateKey) }
```

Comment:
I think that an Openstack AuthKeyPair is the same as an AuthPrivateKey so we might be able to just update https://github.com/ManageIQ/manageiq/pull/18309/files#diff-f9cace1fe4fdcec66f962b23f2110fd6R70

Reply:
You mean change `ManageIQ::Providers::Openstack::InfraManager::AuthKeyPair` to just `AuthKeyPair`? Otherwise I'm not sure what you mean. Also, I thought about changing that line, but was worried about introducing some sort of regression.

Reply:
Yes exactly, rather than add a completely new type I'd rather combine the two.

Reply:
Actually, I just noticed something. It's looking for INFRA AuthKeyPair's. I was thinking Cloud this whole time, duh. Given that there's no AuthKeyPair base class, and the subclass is just an empty subclass, I just removed `authentication_key_pairs` completely as `authentication_private_keys` will pick them up. I can only speculate as to what the plans were for that class, but it looks unnecessary.

Or maybe that's what you meant this whole time and we are now in agreement. Anyway, updated. Please let me know if I've misinterpreted what you want.

Reply:
> Or maybe that's what you meant this whole time and we are now in agreement. Anyway, updated. Please let me know if I've misinterpreted what you want.

Exactly :)  I think it'd be safer to use the old method but just change it to look for `AuthPrivateKey` since the UI is still using authentication_key_pairs

Reply:
Ok, I've modified `authentication_key_pairs` to return `AuthPrivateKey` and removed the separate `authentication_private_keys` method.

Reply:
:+1: